tests/langchain_custom_output_processor_test.py

```diff
@@ -0,0 +1,190 @@
+        Answer: [/INST]
+        """
+    )
+    ragText = """A latte is a coffee drink that consists of espresso, milk, and foam.\
+        It is served in a large cup or tall glass and has more milk compared to other espresso-based drinks.\
+            Latte art can be created on the surface of the drink using the milk."""
+
+    def __format_docs(self, docs):
+        return "\n\n ".join(doc.page_content for doc in docs)
+
+    def __createChain(self):
+
+        resource = Resource(attributes={
+            SERVICE_NAME: "coffee_rag_fake"
+        })
+        traceProvider = TracerProvider(resource=resource)
+        exporter = ConsoleSpanExporter()
+        monocleProcessor = BatchSpanProcessor(exporter)
+
+        traceProvider.add_span_processor(monocleProcessor)
+        trace.set_tracer_provider(traceProvider)
+        self.instrumentor = MonocleInstrumentor()
+        self.instrumentor.instrument()
+        self.processor = monocleProcessor
+        responses =[self.ragText]
+        llm = FakeListLLM(responses=responses)
+        llm.api_base = "https://example.com/"
+        embeddings = HuggingFaceEmbeddings(model_id = "multi-qa-mpnet-base-dot-v1")
+        my_path = os.path.abspath(os.path.dirname(__file__))
+        model_path = os.path.join(my_path, "./vector_data/coffee_embeddings")
+        vectorstore = faiss.FAISS.load_local(model_path, embeddings, allow_dangerous_deserialization = True)
+
+        retriever = vectorstore.as_retriever()
+
+        rag_chain = (
+                {"context": retriever| self.__format_docs, "question": RunnablePassthrough()}
+                | self.prompt
+                | llm
+                | StrOutputParser()
+        )
+        return rag_chain
+
+    def setUp(self):
+        os.environ["HTTP_API_KEY"] = "key1"
+        os.environ["HTTP_INGESTION_ENDPOINT"] = "https://localhost:3000/api/v1/traces"
+
+
+    def tearDown(self) -> None:
+        return super().tearDown()
+
+    @parameterized.expand([
+        ("1", AZURE_ML_ENDPOINT_ENV_NAME, AZURE_ML_SERVICE_NAME),
+        ("2", AZURE_FUNCTION_WORKER_ENV_NAME, AZURE_FUNCTION_NAME),
+        ("3", AZURE_APP_SERVICE_ENV_NAME, AZURE_APP_SERVICE_NAME),
+        ("4", AWS_LAMBDA_ENV_NAME, AWS_LAMBDA_SERVICE_NAME),
+    ])
+    @patch.object(requests.Session, 'post')
+    def test_llm_chain(self, test_name, test_input_infra, test_output_infra, mock_post):
+        app_name = "test"
+        wrap_method = MagicMock(return_value=3)
+        setup_monocle_telemetry(
+            workflow_name=app_name,
+            span_processors=[
+                BatchSpanProcessor(HttpSpanExporter("https://localhost:3000/api/v1/traces"))
+            ],
+            wrapper_methods=[
+                WrapperMethod(
+                    package="langchain_core.retrievers",
+                    object_name="BaseRetriever",
+                    method="invoke",
+                    wrapper=task_wrapper,
+                    output_processor=["entities.json"]
```

Comment:
where is this entities.json loaded from ?

Reply:
added entities.json from test folder